hw2/01-coins.js

```diff
@@ -1,7 +1,51 @@
-/** Exercise 01 - Coins **/
+/* eslint-disable no-console */
+/** Exercise 01 - Coins * */
 
 const calculateChange = (input) => {
-  // Add your code here
+  let ret = `$${input} => `;
+  const dollars = Math.floor(input);
+  if (dollars > 10) {
+    ret += 'Error: the number is too large';
+  } else {
+    const cents = Math.round((input - dollars) * 100);
+    const quarters = Math.floor(cents / 25);
+    const dimes = Math.floor((cents - quarters * 25) / 10);
+    const nickels = Math.floor((cents - quarters * 25 - dimes * 10) / 5);
+    const pennies = Math.floor(cents - quarters * 25 - dimes * 10 - nickels * 5);
+    const changeParts = [];
+    let curDesc = '';
+    if (dollars > 0) {
+      curDesc = 'dollar';
+      if (dollars > 1) curDesc += 's';
+      changeParts.push(`${dollars} ${curDesc}`);
+    }
+    if (quarters > 0) {
+      curDesc = 'quarter';
+      if (quarters > 1) curDesc += 's';
+      changeParts.push(`${quarters} ${curDesc}`);
+    }
+    if (dimes > 0) {
+      curDesc = 'dime';
+      if (dimes > 1) curDesc += 's';
+      changeParts.push(`${dimes} ${curDesc}`);
+    }
+    if (nickels > 0) {
+      curDesc = 'nickel';
+      if (nickels > 1) curDesc += 's';
+      changeParts.push(`${nickels} ${curDesc}`);
+    }
+    if (pennies > 0) {
+      curDesc = 'penny';
+      if (pennies > 1) curDesc = 'pennies';
+      changeParts.push(`${pennies} ${curDesc}`);
+    }
+    if (changeParts.length === 0) {
```

Comment:
Nice edge case

Reply:
If I pass a negative number in this function it's still calculating the changes, maybe you can update that.

Reply:
I like how you just added a letter for plural words. Nice work!

Reply:
Yeah, the function is getting executed for the negative numbers.

Reply:
Also, think about scaling this solution. How many more changes you would have to do if the input is more than 10 and we allow bills 1,2,5,10 and 100 as well? Or say the coins get changed to 1,2,4,8,16.